### PR TITLE
meta with proc & registered_exception? helper

### PR DESCRIPTION
**Pull request by BjornMelgaard**

### CODE REVIEW

File: lib/jsonapi_errorable.rb [hunk 1]
```diff
@@ -55,6 +55,10 @@ def default_exception_handler
     self.class.default_exception_handler
   end
 
+  def registered_exception?(e)
```
Comment (richmolj) [inline lib/jsonapi_errorable.rb]: This code appears unused/untested, what's the intent?

**Comment by richmolj** [comment]
LGTM aside from comment on `#registered_exception?`. Thanks for submitting! 👏

File: lib/jsonapi_errorable.rb (hunk 1 above)
Comment (BjornMelgaard) [inline lib/jsonapi_errorable.rb]: I tested it [here](https://github.com/BjornMelgaard/ngrx-todolist/blob/with_jsonapi_suite/app/controllers/concerns/jsonapi_errors.rb#L33) )

**Comment by BjornMelgaard** [comment]
Testing is right thing, tested)

**Comment by richmolj** [comment]
I'm still hung up a bit on `#registered_exception?`. I see the pattern you would like to do, but kinda disagree with it. That's fine, actually good to see a variety of uses, whatever floats your ⛵️ . But makes me think `#registered_exception?` should be in app code rather than library code.

**Comment by BjornMelgaard** [comment]
I decided to add it here because `self.class._errorable_registry` is private variable

**Comment by richmolj** [comment]
Hm, yeah, the more I think about it I think I'm OK with this. Thanks!

**Comment by stevenharman** [comment]
I *think* I was just looking for something like `#registered_exception?`, but want to verify that the use cases y'all have in mind are in line with what I need to do. In our `ApiController` we report exceptions to a third party error tracking service (App Signal in our case, but it's the same pattern regardless). We're reporting "too much" b/c we're sending error reports for records not found, unauthorized attempts, and several custom error cases. e.g.:

```ruby
  register_exception ActiveRecord::RecordNotFound, status: 404, message: ->(_e) { DEFAULT_NOT_FOUND_MESSAGE }
  register_exception JsonapiCompliable::Errors::RecordNotFound, status: 404, message: ->(_e) { DEFAULT_NOT_FOUND_MESSAGE }
  register_exception ActionPolicy::Unauthorized, status: 403, message: ->(_e) { DEFAULT_NOT_AUTHORIZED_MESSAGE }
  register_exception Billing::OverDeviceLimitError, status: 403, message: true, meta: ->(e) { e.meta }
  register_exception Billing::SubscriptionNotActiveErrror, status: 403, message: true, meta: ->(e) { e.meta }
  register_exception Billing::SubscriptionAlreadyExistsError, status: 422, message: true, meta: ->(e) { e.meta }
```

I think I can avoid reporting any of those to our error tracker by adjusting our `ApiController#rescue_from` to check `#registered_exception?` like this:

```ruby
rescue_from Exception do |e|
  Appsignal.set_error(e) unless registered_exception?(e)
  handle_exception(e)
end
```

Is that consistent with the use cases you have in mind? Is there a "more correct" way to do this?

**Pull request merged**